### PR TITLE
Pin conda-build to version 1.x

### PR DESCRIPTION
**Pull request by jjhelmus**
should address conda-forge/conda-smithy#282

### CODE REVIEW

**Comment by jakirkham** [comment]
Please bump version to `3.1.2` (i.e. the patch number).

**Comment by jakirkham** [comment]
LOL. This is [fun](https://circleci.com/gh/conda-forge/conda-forge-build-setup-feedstock/77). 😒

```
Using Anaconda API: https://api.anaconda.org
Traceback (most recent call last):
  File "/feedstock_root/ci_support/upload_or_check_non_existence.py", line 124, in <module>
    main()
  File "/feedstock_root/ci_support/upload_or_check_non_existence.py", line 104, in main
    exists = built_distribution_already_exists(cli, meta, owner)
  File "/feedstock_root/ci_support/upload_or_check_non_existence.py", line 24, in built_distribution_already_exists
    fname = bldpkg_path(meta)
TypeError: bldpkg_path() missing 1 required positional argument: 'config'
```

I think we are going to have to do some pinnings in the CI scripts too. It's ugly, but tolerable for now.

File: recipe/run_conda_forge_build_setup_win [hunk 1]
```diff
@@ -11,5 +11,8 @@ conda config --set add_pip_as_python_dependency false
 conda update -n root --yes --quiet conda
 conda install -n root --yes --quiet jinja2 conda-build anaconda-client
 
+# KLUDGE to work around changes in conda-build 2.0.0
```
Comment (jakirkham) [inline recipe/run_conda_forge_build_setup_win]: `#` -> `::`

**Comment by jjhelmus** [comment]
> I think we are going to have to do some pinnings in the CI scripts too. It's ugly, but tolerable for now.

Agreed, Circle seems do-able, but how about Travis and AppVeyor?

**Comment by jakirkham** [comment]
Would just add a line after [here](https://github.com/conda-forge/conda-forge-build-setup-feedstock/blob/2e4e4b40e271e369bef950878f85e4d5aca039f2/.travis.yml#L30) in `.travis.yml` and after [here](https://github.com/conda-forge/conda-forge-build-setup-feedstock/blob/2e4e4b40e271e369bef950878f85e4d5aca039f2/appveyor.yml#L58) in `appveyor.yml`.

Comments referencing this PR would be nice.

**Comment by jjhelmus** [comment]
Oh wait, this recipe doesn't use itself, so patching is not as hard as I though.

**Comment by jakirkham** [comment]
> Oh wait, this recipe doesn't use itself, so patching is not as hard as I though.

Yeah, I have been waiting for that to bite us, but we are lucky thus far. Will give it some more thought. Admittedly the same kludgy things we are doing here should still work in that case. Though ideally we would bootstrap this somehow.

**Comment by jakirkham** [comment]
LGTM. Will merge when it passes, thanks for putting this together.

**Comment by jjhelmus** [comment]
No problem.  I'll add the issue from conda-smithy to this repo so as a reminder to remove this kludge once the upload script is fixed.

**Comment by jakirkham** [comment]
Also opened issue ( https://github.com/conda-forge/conda-forge-build-setup-feedstock/issues/29 ) to discuss how we handle building bootstrapping `conda-forge-build-setup`.

**Comment by asmeurer** [comment]
Checks pass. Should this get merged?

**Comment by asmeurer** [comment]
And once this is merged, will I need to wait for my feedstock to be re-rendered, or will it just work?

**Comment by jakirkham** [comment]
Awesome. Thanks for the message.

Nope. Should just work as that feedstock already uses `conda-forge-build-setup`.

**Pull request merged**